src/screens/ContactScreen/index.js

```diff
@@ -128,35 +131,36 @@ const ContactScreen = ({navigation}) => {
           })) || [];
 
         const dataUser = [...followedUsers, ...unfollowedUsers];
-        setUsers(
+        setUsersSearch(
           dataUser?.filter((item) =>
             isAddParticipant ? !existParticipants.includes(item.username) : item
           )
         );
       }
-      setLoading(false);
+      setIsLoadingSearchResult(false);
     } catch (error) {
-      setLoading(false);
+      setIsLoadingSearchResult(false);
     }
   };
 
   const debounced = React.useCallback(
     debounce((changedText) => {
-      handleSearch();
-      setDebouncedText(changedText);
+      handleSearch(changedText);
     }, 1000),
     []
   );
 
   React.useEffect(() => {
-    try {
-      setLoading(true);
-      getDiscoveryUser();
-      setLoading(false);
-    } catch (error) {
-      setLoading(false);
+    if (text.length === 0) {
+      try {
+        setLoading(true);
+        getDiscoveryUser();
+        setLoading(false);
+      } catch (error) {
+        setLoading(false);
+      }
     }
-  }, []);
+  }, [text]);
 
   React.useEffect(() => {
     if (users.length > 0) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [115-143]

The `handleSearch` function now takes a parameter `searchText`. This is an improvement over the previous version where the text was directly accessed from the state. This makes the function more pure and easier to test as it doesn't rely on external state. However, error handling could be improved. Currently, if an error occurs, `setIsLoadingSearchResult(false)` is called twice - once in the catch block and once after the data fetch. Consider refactoring this to avoid redundancy.

```diff
  const handleSearch = async (searchText) => {
     try {
      setIsLoadingSearchResult(true);
 
       const cancelToken = cancelTokenRef?.current?.token;
      const data = await DiscoveryRepo.fetchDiscoveryDataUser(searchText, isAnon, {cancelToken});
       if (data.success) {
         const followedUsers =
           data?.followedUsers?.map((item) => ({
...
         );
       }
- 140:       setIsLoadingSearchResult(false);
     } catch (error) {
      // Handle error appropriately here
     }
+ 144:     setIsLoadingSearchResult(false);
   };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->